plugins/reposync.py

```diff
@@ -186,15 +212,31 @@ def metadata_target(self, repo):
 
     def pkg_download_path(self, pkg):
         repo_target = self.repo_target(pkg.repo)
+
+        pkg_location = pkg.location
         pkg_download_path = os.path.realpath(
-            os.path.join(repo_target, pkg.location))
+            os.path.join(repo_target, pkg_location))
+
+        # If we haven't set the blobstore option (i.e. preserve full path)
+        # and we see references to a file outside our destination,
+        # then we should just store the package in the target location
+        if not self.opts.blobstore:
+            if not pkg_download_path.startswith(os.path.join(repo_target, '')):
+                pkg_location = os.path.basename(pkg_location)
+                pkg_download_path = os.path.realpath(
+                    os.path.join(repo_target, pkg_location))
+
         # join() ensures repo_target ends with a path separator (otherwise the
         # check would pass if pkg_download_path was a "sibling" path component
         # of repo_target that has the same prefix).
```

Comment:
This commend now lost it's meaning. It should be also moved (together with `os.path.join(repo_target, '')` move)